fea-rs/src/compile/compile_ctx.rs

```diff
@@ -1129,18 +1130,29 @@ impl<'a> CompilationCtx<'a> {
             return (0, None);
         };
 
-        let locations = metric
-            .location_values()
-            .map(|loc_value| {
-                let user_loc = loc_value
-                    .location()
-                    .items()
-                    .map(|axis_value| (axis_value.axis_tag().to_raw(), axis_value.value().parse()))
-                    .collect();
-                let value = loc_value.value().parse_signed();
-                (user_loc, value)
-            })
-            .collect::<HashMap<_, _>>();
+        let mut locations = HashMap::new();
+        for metric_loc in metric.location_values() {
+            let mut pos = NormalizedLocation::new();
+            for axis_value in metric_loc.location().items() {
+                let tag = axis_value.axis_tag().to_raw();
+                let Some((_, axis)) = var_info.axis(tag) else {
+                    self.error(metric.range(), format!("unrecognized tag '{tag}'"));
```

Comment:
We don't need to error here, since we've already checked this condition in the validation pass. The idea that once you get to the compile pass you can (generally) just focus on building up the data structures.

There are a few exceptions regarding things that are not available in the validation pass, but this is not one of them.

Reply:
Added a note it's definitely valid and unwrapped